Make OTEL_DOTNET_AUTO_EXCLUDE_PROCESSES consistent (#2765)
open-telemetry/opentelemetry-dotnet-instrumentation: docs/config.md

| `OTEL_DOTNET_AUTO_FAIL_FAST_ENABLED` | Enables possibility to fail process when automatic instrumentation cannot be executed. It is designed for debugging purposes. It should not be used in production environment. \[1\]                                                    | `false`       | [Experimental](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/versioning-and-stability.md) |
| `OTEL_LOG_LEVEL`                     | SDK log level. (supported values: `none`,`error`,`warn`,`info`,`debug`)                                                                                                                                                                 | `info`        | [Stable](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/versioning-and-stability.md)       |

\[1\] If `OTEL_DOTNET_AUTO_FAIL_FAST_ENABLED` is set to `true` then processes
excluded from instrumentation by `OTEL_DOTNET_AUTO_EXCLUDE_PROCESSES` will fail
instead of silently continue.

> \[2\] `OTEL_DOTNET_AUTO_EXCLUDE_PROCESSES` can NOT be used in case process is
> not started directly from exe. (e.g.: `dotnet MyApp.dll` has process name `dotnet`)

Reviewer: To many negations. Hard to read.

To many negations. I find it hard to read. Maybe something like

```suggestion
\[2\] `OTEL_DOTNET_AUTO_EXCLUDE_PROCESSES` only applies to the initial process name (e.g. for `dotnet MyApp.dll` has process name `dotnet`).
```

@theletterf Thoughts?

Author: I agree, hoped that everybody else here understood and offered something better for docs 😆 

`initial process name` seems also confusing to me. For example I think `dotnet run` spins 2 processes: `dotnet` and `MyApp.exe`

Reviewer 2: I think a "notice" style works better in this case.
```markdown
\[2\] Notice that applications launched via `dotnet MyApp.dll` have process name `dotnet` or `dotnet.exe`.
````

Author: https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/2765/commits/153066b1fd3b6ba63c28d5912dad6ced32cd54db
